demo/grid-styling-demos.html

```diff
@@ -0,0 +1,143 @@
+
+    <vaadin-demo-snippet id='grid-styling-demos-cell-class-generator'>
+      <template preserve-content>
+
+        <vaadin-grid aria-label="Styling Grid With Cell Class Generator Example" id="grid" size="200">
+          <vaadin-grid-column path="name.first" header="First name"></vaadin-grid-column>
+          <vaadin-grid-column path="name.last" header="Last name"></vaadin-grid-column>
+          <vaadin-grid-column path="email"></vaadin-grid-column>
+        </vaadin-grid>
+
+        <script>
+          window.addDemoReadyListener('#grid-styling-demos-cell-class-generator', function(document) {
+            const grid = document.querySelector('vaadin-grid');
+            grid.items = Vaadin.GridDemo.users;
+
+            grid.cellClassNameGenerator = function(rowData, column) {
+              let classes = rowData.item.gender;
+              if (column.path === 'email') {
+                classes += ' email';
+              }
+              return classes;
+            };
+          });
+        </script>
+
+        <!-- Place in index.html or in a separate HTML import -->
+        <dom-module id="my-grid-styles" theme-for="vaadin-grid">
+          <template>
+              <style>
+                /* Background needs a stronger selector to not be overridden */
+                [part~="cell"].female {
+                  background: rgb(255, 240, 240);
+                }
+
+                [part~="cell"].male {
+                  background: rgb(245, 245, 255);
+                }
+
+                .email {
+                  font-style: italic;
+                }
+              </style>
+          </template>
+        </dom-module>
+      </template>
+    </vaadin-demo-snippet>
+
+    <h3>Updating Styles After Changes in the Data</h3>
+
+    <vaadin-demo-snippet id='grid-styling-demos-regenerate-classes'>
+      <template preserve-content>
+
+        <vaadin-grid aria-label="Updating Cell Class Generator Example" id="grid" size="200">
+          <vaadin-grid-column id="check" flex-grow="0" width="60px"></vaadin-grid-column>
+          <vaadin-grid-column path="name.first" header="First name"></vaadin-grid-column>
+          <vaadin-grid-column path="name.last" header="Last name"></vaadin-grid-column>
+          <vaadin-grid-column path="email"></vaadin-grid-column>
+        </vaadin-grid>
+
+        <script>
+          window.addDemoReadyListener('#grid-styling-demos-regenerate-classes', function(document) {
+            const grid = document.querySelector('vaadin-grid');
+            grid.items = Vaadin.GridDemo.users;
+
+            const checkColumn = document.querySelector('#check');
+            checkColumn.renderer = function(root, column, rowData) {
+              if (!root.firstElementChild) {
+                root.innerHTML = '<vaadin-checkbox></vaadin-checkbox>';
+                root.firstElementChild.addEventListener('checked-changed', function(e) {
+                  root.item.checked = e.detail.value;
```

Comment:
Rather than changing the provided data (not necessarily ok), we could have a `const checkedItems = []` array or an object in the demo code which would host the state of the checked items. Currently, as the data comes from a shared source, any changes to it are effective in the other demos as well.

Reply:
Done